diff.js

```diff
@@ -41,12 +44,18 @@ import { equalityStrict } from './function.js'
 export const simpleDiffString = (a, b) => {
   let left = 0 // number of same characters counting from left
   let right = 0 // number of same characters counting from right
+  let inSurrogate = false
   while (left < a.length && left < b.length && a[left] === b[left]) {
+    inSurrogate = highSurrogateRegex.test(a[left])
```

Comment:
Is it necessary to test this in every iteration? Couldn't we perform all the checks at the very end of the function and then adjust `left` & `right` if necessary?